application/server.py

```diff
@@ -86,13 +107,23 @@ def handle_join_room_form():
         room_name = request.form.get('room-name')
         room_password = request.form.get('room-password')
 
-        # TODO: Look for room in database
-
-        if room_password == "correct": # TODO: Verify password is correct
-            # TODO: create a new player in the database (you'll need the game's id)
-            return redirect('/waiting-room')
-        else:
-            return redirect('/enter-room')
+        games = models.Game.query.all()
+        
+        for game in games:
+            game = models.Game.query.filter_by(room_name=room_name, room_password=room_password).first()
+            if game.room_name == room_name:
```

Comment:
There if statements are not necessary. If you're only querying for games that match your room name and password, they you won't ever get games that don't match that.